### PR TITLE
nt740/feature/public reading log

### PR DESCRIPTION
**Pull request by mekarpeles**
Allows users to manage their reading logs to be publicly shareable.
- /accounts/books takes use to internal reading log page
- /account/notifications allows them to manage their privacy settings
- /people/{username}/books shows an error msg if set to private, otherwise, renders a safe version of reading log page if public

### CODE REVIEW

File: openlibrary/plugins/upstream/account.py [hunk 1]
```diff
@@ -731,26 +757,52 @@ def get_works(self, key):
...
+        user_preferences = web.ctx.site.get('/people/%s/preferences' % username)
+        notifications = (user_preferences or {}).get('notifications')
+        if notifications and notifications.get('public_readlog', 'yes') == 'yes':
+            readlog = ReadingLog()
+            works = readlog.get_works(key)
+            return render['account/books'](
+                works, key, reading_log=readlog.reading_log_counts,
+                lists=readlog.lists, username=username)
+        return render['generic'](
+            '<h3>Sorry, this user has choosen not to make their reading log public.</h3>' \
```
Comment (LeadSongDog) [inline openlibrary/plugins/upstream/account.py]: Should read "Sorry, this user has not chosen to make their reading log public." It's a small difference, but reflects a default stance of privacy.

**Comment by mekarpeles** [comment]
Addresses #583

**Comment by salman-bhai** [comment]
Hey Mek, could you use the keyword **Closes** rather than __Addresses__ It'll automatically close the Issue when you merge the PR!

**Comment by mekarpeles** [comment]
@salman-bhai this issue hasn't been closed until the pull request is merged upstream (so `addresses` was intentional)

File: openlibrary/templates/type/user/view.html [hunk 2]
```diff
@@ -37,6 +38,12 @@ <h1>$page.displayname</h1>
 
     $:sanitize(format(page.description))
 
+    <h2>Reading Log</h2>
+    $if settings.get('public_readlog') == "yes":
+      <p>This user has chosen to publicly share the books they are <a href="$(page.key)/books/currently-reading">currently reading</a>, <a href="$(page.key)/books/already-read">have already read</a>, and <a href="$(page.key)/books/want-to-read">want to read</a>!</p>
+    $else:
+      <p>This user has chosen to keep their Reading Log private</p>
```
Comment (bfalling) [inline openlibrary/templates/type/user/view.html]: QUESTION: Is this a downer to articulate that someone has made their Reading Log private? Should we show nothing here? Not sure, just curious...

File: static/css/master.css [hunk 3]
```diff
@@ -6063,3 +6062,15 @@ ul#myreads-paginator li.selected:hover {
 a.btn {
     text-decoration: none;
 }
+
+.generic-msg {
```
Comment (bfalling) [inline static/css/master.css]: SUGGESTION: Use a more meaningful selector name.

File: openlibrary/templates/account/notifications.html [hunk 4]
```diff
@@ -24,10 +22,8 @@
         <p class="sansserif larger">$_("Notifications are connected to Lists at the moment. If one of the books on your list gets edited, or a new book comes into the catalog, we'll let you know, if you want.")</p>
         <br/>
 
-
-
         <div class="formElement">
-            <h3 class="collapse">$_("Would you like to receive remarkably low traffic update-style emails from Open Library?")</h3>
+            <h3 class="collapse">$_("Would you like to receive very occassional emails from Open Library?")</h3>
```
[the remark this answers is not in the document]
Comment (mekarpeles) [reply openlibrary/templates/account/notifications.html]: resolved

File: openlibrary/templates/account/books.html [hunk 5]
```diff
@@ -1,18 +1,40 @@
...
+        $username &raquo; $header_title
+      $else:
+        $header_title
+    </h1>
+
+    $if owners_page:
+      <h3>
+        Your reading log is currently set to $("public" if public else "private") &mdash;          
+        $if public:
+          <button onclick="prompt('Copy to clipboard: Ctrl+C, Enter', window.location.protocol + '//' + window.location.hostname + '/people/$(username)/books/$(key)');">Share</button> or
```
[the remark this answers is not in the document]
Comment (mekarpeles) [reply openlibrary/templates/account/books.html]: resolved

File: openlibrary/templates/type/user/view.html (hunk 2 above)
Comment (mekarpeles) [reply openlibrary/templates/type/user/view.html]: Hmm, I wondered this too. It was designed this in response to @yonial's feedback -- if a user tries to go to a private reading log, they get re-routed to the user's profile page (and we show this message). Agreed it's not perfect

File: static/css/master.css (hunk 3 above)
Comment (mekarpeles) [reply static/css/master.css]: I think this can actually be removed! Thanks!

File: openlibrary/templates/type/user/view.html (hunk 2 above)
Comment (bfalling) [reply openlibrary/templates/type/user/view.html]: Understood. I tend to think that eventually, the Reading Log should be a part of the user's profile page, shown or hidden depending on the user's settings. Anyway, fine to proceed.

Comment (mekarpeles) [reply openlibrary/templates/type/user/view.html]: agreed, there's actually a design + path to replace this, so will keep this in consideration

**Comment by mekarpeles** [comment]
Closes #740

File: openlibrary/templates/type/user/view.html (hunk 2 above)
Comment (hornc) [reply openlibrary/templates/type/user/view.html]: @mekarpeles I agree with @LeadSongDog 's comment on an earlier version of this text, about the subtle change to make it clear the privacy the default.  This version may not be quite accurate either, as the user has not necessarily chosen anything at this point: "This user has not chosen to share their Reading Log"  would be more accurate, or if you want less of a downer, maybe "This user has not yet chosen to share their Reading Log" . The ''yet'' may be too much / hopeful / misleading though, I don't have a strong opinion :)  I think there should be a way forward that is both more accurate reflecting default privacy (which is a good thing), and simultaneously less of a downer.

**Pull request merged**